modules/nixosModules.nix

```diff
@@ -15,7 +15,11 @@ in
       nixosModules = mkOption {
         type = types.lazyAttrsOf types.deferredModule;
         default = { };
-        apply = mapAttrs (k: v: { _file = "${toString moduleLocation}#nixosModules.${k}"; imports = [ v ]; });
+        apply = mapAttrs (k: v: {
+          _file = "${toString moduleLocation}#nixosModules.${k}";
+          key = "${toString moduleLocation}#nixosModules.${k}";
```

Comment:
```suggestion
          # Note: this neglects to represent potential differences due to input
          #       overrides or flake-parts extendModules. However, the cost for this
          #       is too high or plain infeasible respectively. We choose to implement
          #       deduplication and disabledModules regardless, because not doing
          #       so poses a more direct problem.
          key = "${toString moduleLocation}#nixosModules.${k}";
```